submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,241 @@
+zone 1 it programs OVN db in zone 1. So only pods in zone 1 nodes will be able to talk to other clusters. Pods in zone 2 or zone 3 will not
+be able to reach remote clusters connected via Submariner.
+
+As part of the proposal, we plan to support both the modes and OVN cluster deployments where interconnect
+is not enabled as well.
+
+### Route APIs
+
+As part of this proposal, we are planning to add two new CRDs in Submariner. The plan is to decouple OVN dataplane programming so that this
+can be moved to OVN, when OVN provides a way to add custom routes. We can propose this CRD or consume the CRD that they provide at a later
+point of time.
+
+#### SubmarinerGwRoute
+
+This CR will be created when a remote endpoint is added and there will be one CR per remote cluster. This crd has two fields
+
+* NextHops - Specifies the list of next hop to reach the remote cluster, in this case it will be the IP of ovn-k8s-mp0
+  interface, the interface used by OVN for host networking.
+* RemoteCIDR - Specifies the list of  remote CIDRs reachable via this cluster.
+
+This CR will be used by the route agent pod running on the non-Gateway node to program OVN to send the traffic destined to
+remote clusters via the Submariner tunnel.
+
+``` go
+type SubmarinerGwRoute struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    SubmarinerRoutePolicySpec SubmarinerRoutePolicySpec `json:"submarinerRoutePolicySpec"`
+}
+
+type SubmarinerRoutePolicySpec struct {
+    //Specifies the list of next hops to reach the remote CIDRs
+    NextHops []string `json:"nextHops"`
+
+    //Specifies the remote CIDRs available via the next hop
+    RemoteCidr []string `json:"remoteCidr"`
+}
+```
+
+#### SubmarinerNonGWRoute
+
+This CR will be created when a remote endpoint is created and there will be one created endpoint.
+
+* NextHops - Specifies the list of next hops. In this case it will be the transit switch IP.
+* RemoteCIDR - Specifies the list of remote CIDRs reachable via this gateway.
+
+This CR will be used by the RouteAgent pods running on the non-Gateway nodes to
+
+* In non-g/w node - If the route-agent pod is not in the same zone as Gateway node zone, send the traffic to the g/w node zone.
+* In g/w node - add route to send the traffic from other zones, destined to remote cluster to the submariner tunnel
+
+``` go
+type SubmarinerNonGWRoute struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    SubmarinerRoutePolicySpec SubmarinerRoutePolicySpec `json:"submarinerRoutePolicySpec"`
+}
+
+type SubmarinerRoutePolicySpec struct {
+    //Specifies the list of next hops to reach the remote CIDRs
+    NextHops []string `json:"nextHops"`
+
+    //Specifies the remote CIDRs available via the next hop
+    RemoteCidr []string `json:"remoteCidr"`
+}
+```
+
+## Design Details
```

Comment:
I think we should also cover GW node switchover under Design Details (e.g: SM with 2 GW HA  setup whereas GW1 node is assigned to OVN-Zone1 and GW2  node is assigned to OVN-Zone2. )

Reply:
Done.